### ISSUE TITLE
java.lang.NoClassDefFoundError while initializing OfflineCredentials forApi(Api.ADWORDS)

### ISSUE THREAD

**Issue by ragidayananda** [googleads/googleads-java-lib#156]
I am using maven and this is my pom dependency section and my IDE is intelliJ
 
```
<dependencies>
        <dependency>
            <groupId>com.myapp</groupId>
            <artifactId>dependent-library-of-my-own</artifactId>
            <version>1.0.19-SNAPSHOT</version>
        </dependency>
        <dependency>
            <groupId>com.google.api-ads</groupId>
            <artifactId>ads-lib</artifactId>
            <version>4.1.0</version>
        </dependency>
        <dependency>
            <groupId>com.google.api-ads</groupId>
            <artifactId>adwords-axis</artifactId>
            <version>4.1.0</version>
        </dependency>

    </dependencies>
```

Getting below exception while running an application that tries to create OfflineCredentials for Adwords.

`
java.lang.ClassNotFoundException: org.slf4j.event.LoggingEvent\n\tat java.net.URLClassLoader.findClass(URLClassLoader.java:381)\n\tat java.lang.ClassLoader.loadClass(ClassLoader.java:424)\n\tat sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)\n\tat java.lang.ClassLoader.loadClass(ClassLoader.java:357)\n\t... 98 common frames omitted\nWrapped by: java.lang.NoClassDefFoundError: org/slf4j/event/LoggingEvent\n\tat java.lang.Class.getDeclaredMethods0(Native Method)\n\tat java.lang.Class.privateGetDeclaredMethods(Class.java:2701)\n\tat java.lang.Class.getDeclaredMethods(Class.java:1975)\n\tat com.google.inject.spi.InjectionPoint.getInjectionPoints(InjectionPoint.java:688)\n\tat com.google.inject.spi.InjectionPoint.forInstanceMethodsAndFields(InjectionPoint.java:380)\n\tat com.google.inject.spi.InjectionPoint.forInstanceMethodsAndFields(InjectionPoint.java:399)\n\tat com.google.inject.internal.BindingBuilder.toInstance(BindingBuilder.java:84)\n\tat com.google.api.ads.common.lib.utils.logging.LoggingModule.configure(LoggingModule.java:68)\n\tat com.google.api.ads.adwords.lib.utils.logging.AdWordsLoggingModule.configure(AdWordsLoggingModule.java:51)\n\tat com.google.inject.AbstractModule.configure(AbstractModule.java:62)\n\tat com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:340)\n\tat com.google.inject.AbstractModule.install(AbstractModule.java:122)\n\tat com.google.api.ads.adwords.lib.AdWordsModule.configureLogging(AdWordsModule.java:47)\n\tat com.google.api.ads.adwords.lib.AdWordsModule.configure(AdWordsModule.java:39)\n\tat com.google.inject.AbstractModule.configure(AbstractModule.java:62)\n\tat com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:340)\n\tat com.google.inject.spi.Elements.getElements(Elements.java:110)\n\tat com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:138)\n\tat com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:104)\n\tat com.google.inject.Guice.createInjector(Guice.java:96)\n\tat com.google.inject.Guice.createInjector(Guice.java:73)\n\tat com.google.inject.Guice.createInjector(Guice.java:62)\n\tat com.google.api.ads.adwords.lib.utils.AdWordsInternals.<clinit>(AdWordsInternals.java:37)\n\tat com.google.api.ads.common.lib.auth.OfflineCredentials$Api.getInternals(OfflineCredentials.java:107)\n\tat com.google.api.ads.common.lib.auth.OfflineCredentials$Builder.defaultOptionals(OfflineCredentials.java:318)\n\tat com.google.api.ads.common.lib.auth.OfflineCredentials$Builder.forApi(OfflineCredentials.java:312)\n\tat com.usermind.usermindsdk.external.GoogleAdwordsClient.getAdwordsSession(GoogleAdwordsClient.java:164)\n\tat `

**Comment by nwbirnie** [comment]
Thanks for reporting this issue. 

It's likely that you have a conflicting version of the slf4j-api dependency in your project. Could you try putting 

    <dependency>
      <groupId>org.slf4j</groupId>
      <artifactId>slf4j-api</artifactId>
      <version>1.7.7</version>
    </dependency>

as the first dependency in your pom please?

Additionally, if that works, you should inspect the output of mvn dependency:tree -Dverbose to identify the source of the conflict and ensure that all users of slf4j have a consistent version. In particular if you could post any conflicting versions you find and their source we can take a look at including a more compatible version in our libraries.

**Comment by ragidayananda** [comment]
Thanks for the hint. Issue got resolved after adding the slf4j-api version 1.7.25

Looks like the issue was with the version. as mentioned in this doc, https://www.slf4j.org/codes.html#log4j_version it was required a version 1.7.15 or later.
